integration-tests/tests/src/tests/mixed.ts

```diff
@@ -220,23 +345,882 @@ describe("Mixed", () => {
+          expectMatchingFlatDictionary(created.value);
+        });
+
+        it("can create and access a dictionary with different types (input: Default value)", function (this: RealmContext) {
+          const created = this.realm.write(() => {
+            // Pass an empty object in order to use the default value from the schema.
+            return this.realm.create<IMixedWithDefaultCollections>(MixedWithDefaultCollectionsSchema.name, {});
+          });
+
+          expect(this.realm.objects(MixedWithDefaultCollectionsSchema.name).length).equals(1);
+          expectMatchingFlatDictionary(created.mixedWithDefaultDictionary);
+        });
+
+        it("can create and access a dictionary (input: Spread embedded Realm object)", function (this: RealmContext) {
+          const created = this.realm.write(() => {
+            const { embeddedObject } = this.realm.create<IMixedAndEmbedded>(MixedAndEmbeddedSchema.name, {
+              embeddedObject: { value: 1 },
+            });
+            expect(embeddedObject).instanceOf(Realm.Object);
+
+            // Spread the embedded object in order to use its entries as a dictionary in Mixed.
+            return this.realm.create<IMixedSchema>(MixedSchema.name, {
+              value: { ...embeddedObject },
+            });
+          });
+
+          expect(created.value).instanceOf(Realm.Dictionary);
+          expect(created.value).deep.equals({ value: 1 });
+        });
+
+        it("can create and access a dictionary (input: Spread custom non-Realm object)", function (this: RealmContext) {
+          const created = this.realm.write(() => {
+            class CustomClass {
+              constructor(public value: number) {}
+            }
+            const customObject = new CustomClass(1);
+
+            // Spread the embedded object in order to use its entries as a dictionary in Mixed.
+            return this.realm.create<IMixedSchema>(MixedSchema.name, {
+              value: { ...customObject },
+            });
+          });
+
+          expect(created.value).instanceOf(Realm.Dictionary);
+          expect(created.value).deep.equals({ value: 1 });
+        });
+
+        it("inserts list items via `push()`", function (this: RealmContext) {
+          const created = this.realm.write(() => this.realm.create<IMixedSchema>(MixedSchema.name, { value: [] }));
+          const list = created.value as Realm.List<any>;
+          expect(list.length).equals(0);
+
+          this.realm.write(() => {
+            list.push(...flatListAllTypes);
+            list.push(this.realm.create(MixedSchema.name, unmanagedRealmObject));
+          });
+          expectMatchingFlatList(list);
+        });
+
+        it("inserts dictionary entries", function (this: RealmContext) {
+          const created = this.realm.write(() => this.realm.create<IMixedSchema>(MixedSchema.name, { value: {} }));
+          const dictionary = created.value as Realm.Dictionary<any>;
```

Comment:
Again, it would be great if `chai` just did this, but I think this could be simplified if you defined a function asserting this is a dictionary and used that instead.

```
function expectDictionary(value: unknown): asserts value is Realm.Dictionary {
  expect(value).instanceOf(Realm.Dictionary);
}
````

```suggestion
          const { value: dictionary } = this.realm.write(() => this.realm.create<IMixedSchema>(MixedSchema.name, { value: {} }));
          expectDictionary(dictionary);
```

Reply:
I've applied updates and it was definitely a good idea to use a custom function that asserts, thanks for the suggestion 🙂